pulp_2_tests/tests/rpm/api_v2/test_rich_weak_dependencies.py

```diff
@@ -101,50 +102,92 @@ def test_all(self):
         )
-        repo = client.post(REPOSITORY_PATH, body)
-        self.addCleanup(client.delete, repo['_href'])
-        repo = client.get(repo['_href'], params={'details': True})
-        sync_repo(cfg, repo)
-        publish_repo(cfg, repo)
-        verify = cfg.get_hosts('api')[0].roles['api'].get('verify')
-        sudo = () if cli.is_root(cfg) else ('sudo',)
+        cls.repo = cls.api_client.post(REPOSITORY_PATH, body)
+        cls.repo = cls.api_client.get(
+            cls.repo['_href'], params={'details': True}
+        )
+        sync_repo(cls.cfg, cls.repo)
+        publish_repo(cls.cfg, cls.repo)
+
+    def test_positive_package_manager_installs_rpm(self):
+        """Package manager can consume RPM with rich/weak dependencies."""
+        verify = self.cfg.get_hosts('api')[0].roles['api'].get('verify')
+        sudo = () if cli.is_root(self.cfg) else ('sudo',)
         repo_path = gen_yum_config_file(
-            cfg,
-            baseurl=urljoin(cfg.get_base_url(), urljoin(
+            self.cfg,
+            baseurl=urljoin(self.cfg.get_base_url(), urljoin(
                 'pulp/repos/',
-                repo['distributors'][0]['config']['relative_url']
+                self.repo['distributors'][0]['config']['relative_url']
             )),
-            name=repo['_href'],
+            name=self.repo['_href'],
             enabled=1,
             gpgcheck=0,
             metadata_expire=0,  # force metadata to load every time
-            repositoryid=repo['id'],
+            repositoryid=self.repo['id'],
             sslverify='yes' if verify else 'no',
         )
-        cli_client = cli.Client(cfg)
-        self.addCleanup(cli_client.run, sudo + ('rm', repo_path))
+        self.addCleanup(self.cli_client.run, sudo + ('rm', repo_path))
+
         rpm_name = 'Cobbler'
-        pkg_mgr = cli.PackageManager(cfg)
-        pkg_mgr.install(rpm_name)
-        self.addCleanup(pkg_mgr.uninstall, rpm_name)
-        rpm = cli_client.run(('rpm', '-q', rpm_name)).stdout.strip().split('-')
+        self.pkg_mgr.install(rpm_name)
+        self.addCleanup(self.pkg_mgr.uninstall, rpm_name)
+
+        rpm = self.cli_client.run(('rpm', '-q', rpm_name)).stdout.strip().split('-')
         self.assertEqual(rpm_name, rpm[0])
 
+    def test_positive_pulp_admin_shows_required_fields(self):
+        """Search contents of a richnweak repository matching package name.
+
+        This test targets `Pulp #3929`_ and `Pulp Smash #901`_. The
+        `repository content`_ documentation describes the CLI content syntax.
+
+        .. _Pulp #3929:  https://pulp.plan.io/issues/3929
+        .. _Pulp Smash #901: https://github.com/PulpQE/pulp-smash/issues/901
+        .. _repository content:
+            https://docs.pulpproject.org/en/latest/user-guide/admin-client/repositories.html#content-search
+
+        Asserts the required fields are present.
+        """
+        if self.cfg.pulp_version < Version('2.17.1'):
```

Comment:
Perhaps create another class and this test can avoid this nested verification of `pulp_version`.